### PR TITLE
Allow new git_name and git_email variables

### PR DESCRIPTION
**Pull request by nmuesch**
### What does this PR do?

Added two new config options for `apigentools push`:
* git-name
* git-email

Also re-ordered the sub flags of `push` to be defined in an alphabetic order. 

### Motivation

Being able to specify a git name and email for the author of the commits pushed by the `push `command. This allows a user to either use whats on the global system or otherwise allows us to modify the gitconfig of the repos that apigentools is committing to. 

### Review checklist (to be filled by reviewers)

- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] Feature or bugfix must have tests
- [ ] Git history [must be clean](https://github.com/DataDog/apigentools/blob/master/CONTRIBUTING.md#commit-messages)

### CODE REVIEW

File: apigentools/cli.py [hunk 1]
```diff
@@ -257,12 +252,28 @@ def get_cli_parser():
...
+    )
     push_parser.add_argument(
         "--skip-if-no-changes",
         help="Skip committing/pushing for all repositories where only .apigentools-info has changed",
         action="store_true",
         default=env_or_val("APIGENTOOLS_SKIP_IF_NO_CHANGES", False, __type=bool),
     )
+    push_parser.add_argument(
+        "--git-email",
+        help="Email of the user to author git commits as",
```
Comment (bkabrda) [inline apigentools/cli.py]: We should add a note about this setting being permanent for the repo (and do the same for the `--git-name` arg).

File: apigentools/commands/push.py [hunk 2]
```diff
@@ -72,6 +72,14 @@ def run(self):
                     if self.args.skip_if_no_changes and self.git_status_empty():
                         log.info("Only .apigentools file changed for language {}, skipping".format(lang_name))
                         continue
+
+                    # Update git config for this repository to use the provided author's email/name
+                    # If not specified, use the setup from the system/global
+                    if self.args.git_email:
+                        run_command(['git', 'config', 'user.email', self.args.git_email])
```
Comment (bkabrda) [inline apigentools/commands/push.py]: Could you also add `dry_run=self.args.dry_run` to both of these commands?

File: apigentools/cli.py [hunk 3]
```diff
@@ -257,12 +252,28 @@ def get_cli_parser():
...
+        default=env_or_val("APIGENTOOLS_GIT_EMAIL", None),
```
Comment (bkabrda) [inline apigentools/cli.py]: I'd prefer if we had `APIGENTOOLS_<name_of_the_env_var_that_actually_configures_git>`, so e.g. `APIGENTOOLS_GITH_AUTHOR_EMAIL`, `APIGENTOOLS_GIT_AUTHOR_NAME`.

Comment (nmuesch) [reply apigentools/cli.py]: Yupp, that makes sense. I changed this to be more consistent with your suggestion

File: apigentools/commands/push.py (hunk 2 above)
Comment (nmuesch) [reply apigentools/commands/push.py]: So it looks like the git config command doesn't actually support a `dry-run` option. I got some weird behaviors/things in my git config while testing. So instead I put it as part of the condition for running the command.

File: apigentools/cli.py (hunk 1 above)
Comment (nmuesch) [reply apigentools/cli.py]: done!

File: apigentools/commands/push.py (hunk 2 above)
Comment (nmuesch) [reply apigentools/commands/push.py]: misunderstood. Didn't see that `run_command` had a dry_run option 😄 Updated

**Pull request merged**